### PR TITLE
Reduce to one row per patient based on cardinality

### PR DESCRIPTION
**Pull request by rebkwok**
Add the aggregation reduce functions (`exists` and `count`)

### CODE REVIEW

File: tests/test_dsl.py [hunk 1]
```diff
@@ -71,6 +71,34 @@ class OldCohort:
     assert_cohorts_equivalent(cohort, OldCohort)
 
 
+def test_exists_aggregation():
```
Comment (benbc) [inline tests/test_dsl.py:91]: These test names are frack-to-bunt.

**Review by benbc — CHANGES_REQUESTED** [review]
I _think_ the need to specify a column to pick with is obviated by sorting, which we plan to decouple from patient reduction in the new DSL (see [here](https://app.shortcut.com/ebm-datalab/story/16/add-a-sorting-operation-to-the-new-dsl)).

File: tests/test_dsl.py (hunk 1 above)
Comment (rebkwok) [reply tests/test_dsl.py:91]: oops :facepalm:

**Review by benbc — APPROVED** [review]
Thanks, Becky. All good if you're happy to remove that.

**Pull request merged**